fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -77,81 +73,192 @@
+            "MERGE INTO " + SIMPLE_SEARCH_TABLE +
+                    " (" + MODIFIED_COLUMN + "," + CREATED_COLUMN + ", " + CONTENT_SIZE_COLUMN + "," +
+                    MIME_TYPE_COLUMN + ", " + FEDORA_ID_COLUMN +
+                    ") KEY (" + FEDORA_ID_COLUMN + ") SELECT " + MODIFIED_COLUMN + ", " + CREATED_COLUMN +
+                    ", " + CONTENT_SIZE_COLUMN + "," + MIME_TYPE_COLUMN + ", " + FEDORA_ID_COLUMN + " FROM " +
+                    SIMPLE_SEARCH_TRANSACTIONS_TABLE + " WHERE " + TRANSACTION_ID_COLUMN + "= :" +
+                    TRANSACTION_ID_PARAM + " AND " + OPERATION_COLUMN + "='add'";
+
+    private static final String UPSERT_COMMIT_SIMPLE_SEARCH_MYSQL_MARIA = "INSERT INTO " + SIMPLE_SEARCH_TABLE +
+            " (" + MODIFIED_COLUMN + "," + CREATED_COLUMN + ", " + CONTENT_SIZE_COLUMN + "," +
+            MIME_TYPE_COLUMN + ", " + FEDORA_ID_COLUMN +
+            ") SELECT " + MODIFIED_COLUMN + ", " + CREATED_COLUMN +
+            ", " + CONTENT_SIZE_COLUMN + "," + MIME_TYPE_COLUMN + ", " + FEDORA_ID_COLUMN + " FROM " +
+            SIMPLE_SEARCH_TRANSACTIONS_TABLE + " a WHERE " + TRANSACTION_ID_COLUMN + "= :" +
+            TRANSACTION_ID_PARAM + " AND " + OPERATION_COLUMN + "='add' " +
+            "ON DUPLICATE KEY UPDATE " + MODIFIED_COLUMN + " = a." + MODIFIED_COLUMN + ", " +
+            CREATED_COLUMN + " = a." + CREATED_COLUMN + ", " +
+            CONTENT_SIZE_COLUMN + " = a." + CONTENT_SIZE_COLUMN + ", " +
+            MIME_TYPE_COLUMN + " = a." + MIME_TYPE_COLUMN;
+
+    private static final String UPSERT_COMMIT_SIMPLE_SEARCH_POSTGRESQL = "INSERT INTO " + SIMPLE_SEARCH_TABLE +
+            " (" + MODIFIED_COLUMN + "," + CREATED_COLUMN + ", " + CONTENT_SIZE_COLUMN + "," +
+            MIME_TYPE_COLUMN + ", " + FEDORA_ID_COLUMN +
+            ") SELECT " + MODIFIED_COLUMN + ", " + CREATED_COLUMN +
+            ", " + CONTENT_SIZE_COLUMN + "," + MIME_TYPE_COLUMN + ", " + FEDORA_ID_COLUMN + " FROM " +
+            SIMPLE_SEARCH_TRANSACTIONS_TABLE + " WHERE " + TRANSACTION_ID_COLUMN + "= :" +
+            TRANSACTION_ID_PARAM + " AND " + OPERATION_COLUMN + "='add' ON CONFLICT (" + FEDORA_ID_COLUMN + ") " +
+            "DO UPDATE SET " + MODIFIED_COLUMN + " = EXCLUDED." + MODIFIED_COLUMN + ", " +
+            CREATED_COLUMN + " = EXCLUDED." + CREATED_COLUMN + ", " +
+            CONTENT_SIZE_COLUMN + " = EXCLUDED." + CONTENT_SIZE_COLUMN + ", " +
+            MIME_TYPE_COLUMN + " = EXCLUDED." + MIME_TYPE_COLUMN;
+
+    private static final String COMMIT_RDF_TYPES =
+            "INSERT INTO " + SEARCH_RDF_TYPE_TABLE + " (" + RDF_TYPE_URI_COLUMN + ")" +
+                    " SELECT distinct " + RDF_TYPE_URI_COLUMN + " FROM " + SEARCH_RESOURCE_RDF_TYPE_TRANSACTIONS_TABLE +
+                    " WHERE " + TRANSACTION_ID_COLUMN + "= :" + TRANSACTION_ID_PARAM + " AND " + RDF_TYPE_URI_COLUMN +
+                    " NOT IN (SELECT " + RDF_TYPE_URI_PARAM + " FROM " + SEARCH_RDF_TYPE_TABLE + ")";
+
```

Comment:
Probably doesn't matter much either way, but this could also be written as an upsert that does nothing on conflict. I suspect it might be slightly faster, but perhaps not worth the additional complexity of writing 4 different queries.